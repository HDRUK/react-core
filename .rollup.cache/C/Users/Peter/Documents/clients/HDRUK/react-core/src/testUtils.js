// @ts-nocheck
import { __assign } from "tslib";
import React from "react";
import { ThemeProvider } from "@emotion/react";
import { render } from "@testing-library/react";
import { DEFAULT_THEME } from "./theme";
export var Providers = function (_a) {
    var children = _a.children;
    return React.createElement(ThemeProvider, { theme: DEFAULT_THEME }, children);
};
export var renderWithProviders = function (ui, options) {
    return render(ui, __assign({ wrapper: Providers }, options));
};
//# sourceMappingURL=testUtils.js.map