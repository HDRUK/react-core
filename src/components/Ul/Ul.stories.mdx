import { Meta, Story, Canvas } from "@storybook/addon-docs";
import Icon from "../Icon";
import { Card, CardBody } from "../Card";
import Ul from "./Ul";
import Li from "../Li";
import { ReactComponent as CheckIcon } from "../../images/icons/check.svg";
import { ReactComponent as DangerIcon } from "../../images/icons/danger.svg";

<Meta title="Molecules/Ul" component={Ul} />

<Canvas>
    <Story name="Basic">
        {props => (
            <Card>
                <CardBody>
                    <Ul {...props}>
                        <Li>List item 1</Li>
                        <Li>List item 2</Li>
                        <Li>List item 3</Li>
                    </Ul>
                </CardBody>
            </Card>
        )}
    </Story>
    <Story name="With icons">
        {props => (
            <Card>
                <CardBody>
                    <Ul listStyleType="none" {...props}>
                        <Li
                            icon={
                                <Icon
                                    svg={<CheckIcon />}
                                    fill="green600"
                                    size="xl"
                                />
                            }>
                            Icon tick
                        </Li>
                        <Li
                            icon={
                                <Icon
                                    svg={<DangerIcon />}
                                    fill="red600"
                                    size="xl"
                                />
                            }>
                            Icon danger
                        </Li>
                    </Ul>
                </CardBody>
            </Card>
        )}
    </Story>
</Canvas>
