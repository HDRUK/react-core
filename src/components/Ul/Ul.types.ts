import { ComponentProps } from "react";
import { ComponentMarginProps } from "../../types";

export interface UlProps extends ComponentProps<"ul">, ComponentMarginProps {
    listStyleType?: string;
}

export type UlStyleProps = Pick<UlProps, "listStyleType">;
