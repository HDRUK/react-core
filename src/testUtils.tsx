import React from "react";
import { ThemeProvider } from "@emotion/react";
import { render, RenderOptions } from "@testing-library/react";
import { DEFAULT_THEME } from "./theme";

export const Providers: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={DEFAULT_THEME}>{children}</ThemeProvider>;
};

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};
