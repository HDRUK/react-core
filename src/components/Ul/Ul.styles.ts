import { css } from "@emotion/react";
import { UlStyleProps } from "./Ul.types";

export const root =
    ({ listStyleType }: UlStyleProps) =>
    () => {
        return css`
            margin: 0;
            padding: 0;
            list-style-position: inside;
            ${listStyleType ? `list-style-type: ${listStyleType};` : ""}
        `;
    };
