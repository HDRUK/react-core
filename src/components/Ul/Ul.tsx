/** @jsxImportSource @emotion/react */
import { cx } from "@emotion/css";
import useCommonStyles from "../../hooks/useCommonStyles";
import * as styles from "./Ul.styles";
import { UlProps } from "./Ul.types";

const Ul = ({
    children,
    mt,
    mb,
    ml,
    mr,
    m,
    mx,
    my,
    className,
    listStyleType,
    ...outerProps
}: UlProps) => {
    const commonStyles = useCommonStyles({
        mt,
        mr,
        ml,
        mb,
        m,
        mx,
        my,
    });

    return (
        <ul
            className={cx("ui-Ul", className)}
            css={[styles.root({ listStyleType }), commonStyles]}
            {...outerProps}>
            {children}
        </ul>
    );
};

export default Ul;
