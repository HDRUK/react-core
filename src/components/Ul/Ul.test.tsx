import { screen, RenderResult } from "@testing-library/react";
import React from "react";
import { renderWithProviders } from "../../testUtils";
import Ul from "./Ul";

let wrapper: RenderResult;

describe("Given the Ul component", () => {
    describe("When it is rendered", () => {
        beforeEach(() => {
            wrapper = renderWithProviders(<Ul>Content</Ul>);
        });

        it("Then should match the snapshot", () => {
            expect(wrapper.asFragment()).toMatchSnapshot();
        });

        it("Then should have the corretc content", () => {
            expect(screen.queryByText("Content")).toBeTruthy();
        });
    });
});
