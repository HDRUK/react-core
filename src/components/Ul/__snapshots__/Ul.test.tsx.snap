// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Ul component When it is rendered Then should match the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ui-Ul css-rswhjr-Ul"
  >
    Content
  </ul>
</DocumentFragment>
`;
